File: NPB-CPP-cadna/cadna_c-3.1.11/examplesC/ex1_cad.cc
#include <cadna.h>
#include <stdio.h>

using namespace std;

int main()
{
  cadna_init(-1);
  printf("------------------------------------------\n");
  printf("|  Polynomial function of two variables  |\n");
  printf("|  with CADNA                            |\n");
  printf("------------------------------------------\n");

  double_st x = 77617.;
  double_st y = 33096.;
  double_st res;

  res=333.75*y*y*y*y*y*y+x*x*(11*x*x*y*y-y*y*y*y*y*y-121*y*y*y*y-2.0)   
    +5.5*y*y*y*y*y*y*y*y+x/(2*y);

  printf("res: %s\n",strp(res));
  printf("Exact result: -0.82739605994682...\n");
  cadna_end();
}

File: NPB-CPP-cadna/cadna_c-3.1.11/examplesFortran/ex1_cad.f90
      program ex1_cad
      use cadna
      implicit none
      type (double_st) :: y,x,res
      call cadna_init(-1)
      print *,'Polynomial function of two variables with CADNA: '
      x=77617e0
      y=33096e0
      res=333.75*y*y*y*y*y*y+x*x*(11*x*x*y*y-y*y*y*y*y*y-121 &
     &   *y*y*y*y-2.0) &
     &   +5.5*y*y*y*y*y*y*y*y+x/(2*y)
      print *,'res = ',str(res)
      call cadna_end()
      end program ex1_cad
